storybookjs/storybook
# Core: Add 'mapping' to support complex arg values

Issue: #13888
Related: #12613

## What I did

This introduces a `mapping` property for fields in `argTypes`, to allow the use of complex arg values. It looks like this:

```jsx
argTypes: {
  label: {
    control: {
      type: 'select',
      options: ['Normal', 'Bold', 'Italic'],
    },
    mapping: {
      Bold: <b>Bold</b>,
      Italic: <i>Italic</i>,
    },
  },
},
```

This works on any control type, but makes most sense when used with a radio/select type. Control options must be simple strings so they can be serialized and synced to the URL. Control options can still be defined as an object, but the only use case is if you want a custom select label that may not be serializable (e.g. outside the `[a-z0-9 _-]` range).

## How to test

- Is this testable with Jest or Chromatic screenshots? Beyond mapping of the initial value, not really
- Does this need a new example in the kitchen sink apps? Updated official-storybook.
- Does this need an update to the documentation? Yes, done.

If your answer is yes to any of these, please make sure to include it in your PR.
